service/s3/unmarshal.go

```diff
@@ -0,0 +1,114 @@
+package s3
+
+import (
+	"encoding/xml"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/internal/sdkio"
+	restlegacy "github.com/aws/aws-sdk-go-v2/private/protocol/rest"
+)
+
+/*
+   GetObject operation doesn't support 200 status code with errors.
+
+*/
+
+// protoGetObjectUnmarshaler defines unmarshaler for ProtoGetObject operation
+type protoGetObjectUnmarshaler struct {
+	output *GetObjectOutput
+}
+
+// unmarshalOperation is the top level method used with a handler stack to unmarshal an operation response
+// This method calls appropriate unmarshal shape functions as per the output shape and protocol used by the service.
+func (u protoGetObjectUnmarshaler) unmarshalOperation(r *aws.Request) {
+	defer r.HTTPResponse.Body.Close()
+	// isRequestError checks if operation response returned a error response
+	if isRequestError(r) {
+		// if startToken is nil, it would mean there is no xml response body
+		if r.HTTPResponse.Body == nil {
```

Comment:
I think `Body` here will always not be nil.  Can use `EOF` error when `decoder.Token()` is called to determine there is no XML error message.

Reply:
Can investigate what to do if Body is nil. This would only happen if the request was not sent at all, but unmarshal was called for some reason. This probably isn't a case we want to handle.